src/System.Management.Automation/engine/runtime/Binding/Binders.cs

```diff
@@ -3959,12 +3959,7 @@ private PSGetIndexBinder(Tuple<CallInfo, PSMethodInvocationConstraints, bool> tu
 
         public override string ToString()
         {
-            return string.Format(CultureInfo.InvariantCulture,
-                                 "PSGetIndexBinder indexCount={0}{1}{2} ver:{3}",
-                                 this.CallInfo.ArgumentCount,
-                                 _allowSlicing ? string.Empty : " slicing disallowed",
-                                 _constraints == null ? string.Empty : " constraints: " + _constraints,
-                                 _version);
+            return string.Create(CultureInfo.InvariantCulture, $"PSGetIndexBinder indexCount={this.CallInfo.ArgumentCount}{(_allowSlicing ? string.Empty : " slicing disallowed")}{(_constraints == null ? string.Empty : " constraints: " + _constraints)} ver:{_version}");
```

Comment:
@iSazonov I think this is way less readable than the original code, don't you think so?

Reply:
I think more about reducing allocations. My suggestions for improvement I outlined in your PR.